### PR TITLE
chore(next): update dependencies

### LINKED ISSUE #3021: Next branch dependency updates for all projects
We should go ahead and update all project dependencies and check for regressions.

This will include a notable update to Astro 5, which now utilizes Vite 6:
https://astro.build/blog/astro-5/

As well as migrating all projects to a minimum of Node 22, to satisfy and upcoming change on Vercel:

Screenshot 2024-12-04 at 1 02 49 PM

Once changes changes are pushed, all contributors should be encouraged to update to use Node 22 on their local machines.

### PR DESCRIPTION
**Pull request by AdrianGonz97**
## Linked Issue

Closes #3021

> NOTE: Alternative to #3022

## Description

Updates dependencies for the following projects:

- Sites
  - [x] v3 docs - next.skeleton.dev
  - [x] v3 Theme Generator - themes.skeleton.dev
- Packages
  - [x] v3 Core Package
  - [x] v3 React Package
  - [x] v3 Svelte Package
  - [x] Necroparser
  - [x] vite-plugin-tw-plugin-watcher
  - ~~v3 Migration CLI~~ ([reminder sent on PR](https://github.com/skeletonlabs/skeleton/pull/2972#issuecomment-2518347267))

## Checklist

Please read and apply all [contribution requirements](https://next.skeleton.dev/docs/resources/contribute).

- [x] Your branch should be prefixed with: `docs/`, `feature/`, `chore/`, `bugfix/`
- [x] Skeleton v3 contributions must target the `next` branch (NEVER `dev` or `master`)
- [x] Documentation should be updated to describe all relevant changes
- [x] Run `pnpm check` in the root of the monorepo
- [x] Run `pnpm format` in the root of the monorepo
- [x] Run `pnpm lint` in the root of the monorepo
- [x] Run `pnpm test` in the root of the monorepo
- [x] If you modify `/package` projects, please supply a Changeset

## Changsets

[View our documentation](http://localhost:4321/docs/resources/contribute/get-started#changesets) to learn more about [Changesets](https://github.com/changesets/changesets). To create a Changeset:

1. Navigate to the root of the monorepo in your terminal
2. Run `pnpm changeset` and follow the prompts
3. Commit and push the changeset before flagging your PR review for review.

### CODE REVIEW

**Comment by endigo9740** [comment]
@Hugos68 @AdrianGonz97 I'd appreciate a review of all current changes. A few notable updates today:

1. I resolved the Astro v5 "flashbang" issue by wrapping the MDX route with `<LayoutRoot>`
2. I've updated all instances of `getCollections` to use `filePath` and `entry.id` from `meta.mdx` -> `meta`
3. I've updated the Svelte component package to use Vitest v3 beta. This is the only version that supports Vite 6.

Note that I did NOT refactor the React components using the Context API to have the Context set outside the component files. This looks to be a new linting rule added in a major release. We can cross that road when we come to this. The linting error is very clear about the change required.

I've also reached out the the Skeleton team and contributor community to do some manual regression testing against the doc site. See that in the `#skeleton-next` channel on Discord.

**Review by endigo9740 — APPROVED** [review]

**Review by TazorDE — APPROVED** [review]
Can't see anything out of place
lgtm

File: .changeset/pink-dolls-obey.md
Comment (AdrianGonz97) [inline .changeset/pink-dolls-obey.md:1]: a changeset shouldn't be necessary for this PR as all of the changes are internal and won't affect the distributed package

Comment (AdrianGonz97) [reply .changeset/pink-dolls-obey.md:1]: besides removing the changeset, this PR is good to go!

Comment (endigo9740) [reply .changeset/pink-dolls-obey.md:1]: @AdrianGonz97 per the Form plugin import. I saw you were moving the `require()` inline in the plugins section. I was receiving an error about this yesterday - but that was after flipping back from this branch to `next` and reinstalling all deps.

I'll double check if the issue persists on this PR this afternoon before merging though.

Comment (AdrianGonz97) [reply .changeset/pink-dolls-obey.md:1]: > per the Form plugin import. I saw you were moving the require() inline in the plugins section. I was receiving an error about this yesterday saying this was not allowed 

yea that was a new lint rule that was added at some point to [`@typescript-eslint`](https://typescript-eslint.io/rules/no-require-imports) that prevents the usage of `require` in esm files

**Pull request merged**